rasa/core/test.py

```diff
@@ -601,6 +601,12 @@ async def _get_e2e_entity_evaluation_result(
     prediction: PolicyPrediction,
 ) -> Optional[EntityEvaluationResult]:
     previous_event = tracker.events[-1]
+
+    if isinstance(previous_event, SlotSet):
+        # UserUttered events with entities can be followed by SlotSet events
+        # if slots are defined in the domain
+        previous_event = tracker.get_last_event_for(UserUttered)
```

Comment:
If there was an ActionExecuted in between then this condition should be wrong (only if there was nothing between user uttered and the current action except slots and bot uttered stuff)
```suggestion
    if isinstance(previous_event, SlotSet):
        # UserUttered events with entities can be followed by SlotSet events
        # if slots are defined in the domain
        previous_event = tracker.get_last_event_for((UserUttered, ActionExecuted))
        if isinstance(previous_event, UserUttered):
             ...
```